tests/dialects/test_snowflake.py

```diff
@@ -17,6 +17,14 @@ def test_snowflake(self):
         )
         self.validate_identity("COMMENT IF EXISTS ON TABLE foo IS 'bar'")
 
+        self.validate_all(
+            "SELECT i, p, o FROM qt QUALIFY ROW_NUMBER() OVER (PARTITION BY p ORDER BY o) = 1",
+            write={
+                "": "SELECT i, p, o FROM qt QUALIFY ROW_NUMBER() OVER (PARTITION BY p ORDER BY o NULLS LAST) = 1",
+                "snowflake": "SELECT i, p, o FROM qt QUALIFY ROW_NUMBER() OVER (PARTITION BY p ORDER BY o) = 1",
+                "sqlite": "SELECT i, p, o FROM (SELECT i, p, o, ROW_NUMBER() OVER (PARTITION BY p ORDER BY o NULLS LAST) AS _w_0 FROM qt) WHERE _w_0 = 1",
```

Comment:
you should have a subquery alias and then qualify everything

Reply:
Is the current approach problematic, or is it just bad practice to leave it unaliased? I didn't qualify on purpose because I was thinking the correspondence between subquery projections and top-level query projections is 1-1, so there shouldn't be any ambiguities, right?

Reply:
bad practice and postgres doesn’t allow it

Reply:
Got it, I'll make the change then, thanks for clarifying.

Reply:
I found a bug where unselected columns participating in the QUALIFY clause would not be pushed down, so external references were invalid. I tried to fix it in my latest commit, but I haven't yet qualified every column reference. If we need to have this, should I just transform every column in the outer selects & outer where clause?

Reply:
not sure what you mean

Reply:
The bug was that before `p` and `o` wouldn't be selected in the inner query [here](https://github.com/tobymao/sqlglot/pull/1288/files#diff-0b04b14c1b4c9192c8ba11f96a80aa0b79b1a2979835a01767f7c778fdf4eaecR86-R87).

Do we need to qualify all columns with subquery alias in the outer select expression?

Reply:
no, because you can always just use the optimizer